databusclient/client.py

```
@@ -1,721 +1,875 @@
+    - auth_url: Keycloak token endpoint URL
+    - client_id: Client ID for token exchange
+    - expected_sha256: The expected SHA256 checksum for validation
+    - validation_mode: Enum (OFF, WARNING, ERROR) to control validation behavior
+    """
+
+    print(f"Download file: {url}")
+    dirpath = os.path.dirname(filename)
+    if dirpath:
+        os.makedirs(dirpath, exist_ok=True)  # Create the necessary directories
+    # --- 1. Get redirect URL by requesting HEAD ---
+    response = requests.head(url, stream=True)
```

Ruff (0.14.2)

Probable use of `requests` call without timeout

(S113)